lib/langchain/llm/openai.rb

```diff
@@ -172,6 +174,10 @@ def summarize(text:)
       complete(prompt: prompt)
     end
 
+    def default_dimension
+      @defaults[:dimension]
```

Comment:
Do you think we can expand this PR to incorporate changes similar to ones that were done in [this PR](https://github.com/andreibondarev/langchainrb/pull/462)?
```
    def default_dimension
        EMBEDDING_SIZES.fetch(defaults[:embeddings_model_name].to_sym)
```

This is the list of embedding dimensions: https://platform.openai.com/docs/models/embeddings

Reply:
@andreibondarev absolutely. 

One issue might be when models support multiple dimension sizes, we would need to use the passed dimension, rather than `default_dimension` when calling `create_default_schema` to create the vector store collection. 

For instance, the `text-embedding-3-small` model supports both 1536 and 512, the later of which I needed, which is the reason this issue became apparent.

Reply:
@prdanelli Would you mind showing some code to illustrate this issue? I'm trying to conceptualize it.

Reply:
To illustrate the initial dimensions issue or the multiple dimensions for models?

Reply:
Both :)

Reply:
It is kind of tricky to make a test as the request responses are mocked, but the spec I included [here](https://github.com/andreibondarev/langchainrb/pull/487/files#diff-e5cc0fcd8e724fd8ee9b499d8385a9a9f6bda124487df754d832336db0e2d5c5R88-R98) was supposed to show this. 

When I was working to implement a vector store using `text-embedding-3-small` embeddings with 512 dimensions, I was only able to return 1536 vectors, no matter where that was specified in the object initialization or the method call. 

The same issue was present when trying to create the index with anything other than the default dimension size. 

Reference: https://openai.com/blog/new-embedding-models-and-api-updates#NewAndImprovedEmbeddingModelTextAsVector-3

```ruby
# From 0.9.2
api_key = "test123"
llm = Langchain::LLM::OpenAI.new(
      api_key:,
      default_options: {
        embeddings_model_name: "text-embedding-3-small",
        dimension: 512,
      }
    )

llm.embed(text: "test").embedding.count
# => 1536
print llm.embed(text: "test").embedding
# => [-0.009877119, 0.0015331338, 0.015642809, -0.054763675, -0.006405938, -0.012950308, 0.009662549, -0.013552487, 0.028683098, 0.007862934, 0.03181166, -0.00654437, 0.0038587903, 0.010153983, 0.014057764, 0.044658143, -0.0598026, -0.002439863, -0.05110907, 0.036518347, 0.03610305, 0.024419393, 0.03995146, -0.04355069, 0.03455261, -0.019560434, -0.013981626, 0.01061773, 0.031202558, -0.041557267, 0.058805887, -0.028876903, -0.0013644198, -0.038733255, 0.05517897, 0.004139115, 0.02357496, 0.01885443, -0.0056687878, -0.003966075, -0.041446522, -0.045018066, 0.011351419, 0.027257249, 0.030012045, -0.018785214, -0.034386493, -0.038677882, 0.04850655, 0.02786635, -0.031783972, -0.0027790212, 0.048229687, 0.07021268, 0.019325098, 0.023921039, -0.035189398, 0.019920357, 0.01854988, 0.0053365515, -0.0022962398, -0.018840587, 0.03840102, 0.0055061304, -0.031977776, -0.060079463, 0.012237384, 0.031479422, 0.00043973772, 0.021526167, 0.04593172, 0.011884382, -0.012811876, 0.016750265, 0.0087281335, 0.019989572, 0.0026994229, 0.018217644, 0.014576883, -0.008049818, 0.020211063, 0.010984574, -0.019657336, 0.0095864115, 0.031174872, -0.012867249, -0.090534486, 0.006250202, 0.0068973713, -0.03649066, -0.011178379, -0.0035646225, 0.015310572, 0.051690485, 0.023339625, -0.027492583, -0.0025506085, -0.01798231, 0.0024364023, -0.0072538336, 0.010430846, 0.009724844, -0.0013592286, 0.017926937, 0.022467503, 0.018660625, -0.031700913, 0.023588803, -0.01247964, 0.033251353, -0.0058660535, -0.020612516, -0.01755317, 0.029458316, 0.044741202, 0.015102925, 0.10980421, -0.10354709, 0.03394351, -0.07132014, 0.027354151, 0.0305104, 0.012541934, -0.022841271, 0.031617854, -0.02102781, -0.006059858, -0.00944798, -0.02292433, 8.727701e-05, -0.0317286, 0.004935099, 0.033057548, 0.009572568, -0.010811535, -0.0083405245, -0.025153084, 0.004502499, -0.05863977, 0.041584954, 0.011309889, 0.024004098, 0.0063782516, 0.0005065744, -0.016238067, -0.028226273, -0.016805638, 0.040283695, 0.022785898, 0.028198587, -0.011476008, 0.005648023, -0.008001366, 0.003675368, -0.053683907, -0.037764233, -0.01578124, -0.01962965, -0.031313304, -0.0017174212, 0.024197903, -0.0049904715, 0.011302968, -0.025429947, -0.025554536, 0.008596623, -0.007883699, -0.017497797, -0.052714884, 0.019283568, 0.026689678, 0.01551822, 0.013635546, -0.012410424, -0.020612516, 0.024004098, -0.02241213, 0.010382395, -0.022370601, 0.0020678272, -0.020100318, 0.050804522, -0.025554536, 0.01885443, 0.015172141, -0.059138127, 0.02418406, 0.0053400123, 0.0030143554, -0.0138293505, -0.006056397, -0.009392607, 0.029209139, 0.038179528, 0.027658701, -0.077355765, 0.018466821, 0.030870322, -0.013600938, 0.040754363, -0.03516171, -0.0029589827, 0.010264728, -0.011489851, 0.0151582975, 0.013614781, -0.023934882, 0.04706686, -0.017400894, -0.024973122, 0.069769695, -0.0014050842, -0.0019415079, 0.002929566, -0.005481905, 0.0075653056, 0.050915267, 0.012645758, -0.008790428, 0.06107617, -0.008077504, 0.01009861, 0.08782122, 0.012244306, -0.0102093555, 0.08355752, 0.011199144, -0.014189274, 0.0013877802, -0.056452546, -0.04294159, 0.01794078, 0.0076483646, -0.0042290958, -0.023851823, 0.026122106, -0.033528216, -0.0041668015, -0.03494022, -0.051552054, -0.037598114, 0.009503352, -0.029652122, -0.03917624, -0.020737104, -0.013026445, 0.024807004, -0.009614098, 0.025665281, 0.015199827, 0.005215423, -0.037515055, -0.037459683, -0.06888373, 0.027312621, -0.010520827, 0.025471477, 0.0024952358, 0.020930909, -0.008125955, -0.05119213, -0.0036165344, -0.04247092, -0.049807813, -0.066115096, 0.0052085016, 0.029319884, -0.01425849, 0.0134071335, -0.044436652, -0.006648194, 0.015130611, 0.018577566, -0.020958595, -0.031230247, 0.041252717, -0.0012182011, 0.018923646, 0.025942145, 0.026772738, -0.042443234, -0.0135594085, -0.0048935693, -0.028489294, -0.011282203, 0.02414253, 0.01096381, -0.04584866, 0.007177696, 0.060910054, -0.012804954, -0.010866907, 0.0012977994, 0.032919116, 0.016182695, 0.022910487, 0.02028028, -0.040062204, 0.00052993477, -0.057200078, 0.03164554, 0.030150477, 0.0053192475, 0.028406234, -0.009364921, 0.045322616, 0.013780899, -0.007994444, 0.003658064, -0.016653363, 0.008368211, 0.019657336, -0.0008154506, 0.0075030113, 0.014466138, -0.050942954, -0.013241015, 0.057310823, 0.03604768, -0.02271668, 0.012514248, -0.022619778, 0.034081943, 0.032420762, -0.01555975, -0.00268731, 0.0015305382, -0.026814267, 0.0143000195, 0.033168294, -0.09540729, -0.027963253, -0.0035352057, -0.008548172, 0.035355516, 0.032254644, 0.009247254, 0.055428147, -0.005855671, 0.041723385, -0.06700106, 0.011323732, -0.0028638109, -0.012085108, 0.04601478, 0.022633621, 0.04521187, -0.024045628, 0.03081495, 0.014189274, 0.04308002, 0.010610809, -0.0022304847, 0.021512324, 0.02149848, -0.05088758, -0.016750265, 0.014922963, 0.032559194, -0.001794424, 0.014113137, 0.04299696, 0.045101125, 0.010299336, 0.04684537, 0.0199342, -0.026578933, -0.01135834, -0.0145353535, 0.0009326852, 0.010375474, 0.034691043, 0.045544107, 0.017193247, -0.034691043, 0.030538086, 0.04277547, 0.06943746, 0.023104291, 0.0457656, 0.028378548, 0.009690235, 0.035217084, -0.011046869, 0.028226273, 0.0025108093, -0.015656652, 0.027035758, -0.0035179018, -0.019837297, 0.051441308, -0.020335652, -0.017096344, -0.07641443, 0.06323571, 0.025111554, 0.018923646, 0.07962605, 0.018217644, -0.002673467, 0.022398287, -0.03934236, -0.041723385, 0.040422127, -0.0030091642, 0.009178038, -0.03219927, -0.01387088, 0.032559194, -0.002592138, -0.02626054, 0.033445157, 0.032863744, -0.0020695576, -0.012604228, -0.022661308, 0.030593459, -0.0037653488, 0.03186703, 0.01092228, -0.033085234, -0.038207214, -0.011732107, 0.015379788, -0.022370601, 0.0016135973, 0.04507344, 0.018868273, 0.03516171, -0.023021232, -0.0023066222, -0.019920357, -0.03588156, 0.0370167, -0.034220375, -0.016265754, 0.007316128, 0.0038345647, -0.005218884, 0.013663232, -0.024087157, -0.0051219817, 0.0152552, -0.021747658, -0.041668013, 0.018563723, -0.03494022, -0.042304803, 0.008617388, -0.019601963, 0.010541592, -0.035549324, -0.013538644, -0.059857972, 0.015739711, -0.053850025, 0.012098951, -0.0033517834, 0.06495227, 0.018093055, 0.036629092, 0.04607015, 0.015656652, 0.048312746, 0.0510537, -0.00602525, 0.026080577, 0.03463567, 0.04269241, -0.04008989, -0.008790428, -0.019961886, 0.046180896, 0.005000854, -0.03516171, -0.014826061, 0.024377864, -0.003571544, 0.0007129245, -0.07857397, -0.03206084, -0.03319598, -0.030925695, -0.016141165, 0.011981284, 0.1145109, -0.03831796, 0.0012147403, -0.019518904, -0.015130611, 0.03319598, 0.0008509238, 0.023325782, -0.01555975, 0.006063319, 0.033971198, 0.027534112, 0.020114161, -0.0010070924, 0.007849091, -0.07242759, -0.02387951, -0.035383202, 0.067111805, 0.02552685, 0.020377181, -0.010915359, -0.01863294, 0.050472286, -0.0031077971, -0.017622385, 0.0034781024, -0.00049402897, -0.0023741077, -0.030953381, -0.013690919, -0.02162307, 0.029513689, -0.008132877, -0.0114414, -0.01421696, 0.020972438, -0.024779318, -0.017331678, -0.020141847, -0.023685705, -0.007572227, -0.102439635, -0.013033367, -0.061906762, -0.03480179, -0.03206084, 0.018439135, -0.025360731, -0.0024329415, -0.0047447546, -0.010147061, 0.010873829, -0.02934757, -0.014369235, -0.0015625505, 0.017857721, -0.0041702623, -0.0031354835, -0.00901192, 0.04069899, 0.00079209026, -0.021166245, -0.013573252, -0.022467503, 0.012528091, -0.042664725, 0.038677882, 0.0037549664, 0.04900491, -0.00097334955, 0.011448322, -0.015712025, 0.017663915, -0.008769663, 0.0040179873, 0.0059041223, 0.019338941, -0.04368912, -0.01300568, 0.01247964, 0.0046928427, -0.020391025, 0.023810294, 0.0034088865, -0.012105873, 0.038290273, -0.0317286, -0.019740395, -0.011822088, -0.0077660317, -6.872929e-05, -0.022509033, 0.0065582134, -0.0033379402, 0.0030455026, -0.0118082445, -0.017691601, 0.021595383, -0.022342915, 0.0029762867, 0.039148554, -0.03801341, -0.040643618, -0.019505061, 0.027146503, 0.03618611, 0.022619778, -0.010610809, 0.058362905, 0.0056895525, 0.01663952, -0.009060371, -0.025596065, -0.014424608, 0.007028882, -0.021429265, 0.056175683, 0.01962965, 0.007329971, 0.013116427, -0.030150477, 0.004256782, -0.019491218, 0.012064343, 0.020307966, -0.0028914972, -0.010223199, -0.03358359, 0.007952915, 0.015753554, 0.019754238, -0.014057764, 0.015573593, -0.017040972, -0.070157304, -0.0023221958, 0.003761888, 0.01690254, 0.008298995, 0.022772053, 0.009897884, 0.006322879, 0.007295363, 0.013725527, -0.040892795, 0.010742319, 0.011289124, -0.035466265, -0.0188129, -0.048008196, -0.006094466, 0.030039731, -0.028600039, -0.017040972, 0.0364076, 0.02778329, 0.015130611, 0.0070842546, 0.03455261, -0.010776927, -0.004980089, -0.024336334, -0.009219567, 0.0019293951, -0.029153766, -0.012154324, -0.020626359, -0.0035196322, 0.033694334, -0.041668013, -0.039674595, -0.036241483, -0.0067381747, -0.010057081, 0.029707495, 0.006046015, 0.018965175, -0.030067418, 0.0010105532, 0.043578375, -0.028959962, -0.017608542, -0.0013012602, -0.011822088, -0.0061567607, 0.0074130306, 0.03635223, -0.034580298, 0.040975854, 0.00028876038, -0.017331678, -0.044962693, -0.042304803, 0.027423367, -0.0029762867, -0.011559067, -0.029984359, -0.005153129, -0.021941463, 0.01751164, -0.0014050842, 0.00033202037, 0.013483271, -0.0038726334, -0.027617171, 0.0059664166, 0.008437427, 0.015462847, -0.015988888, -0.011109163, 0.0031960474, -0.009357999, -0.015725868, -0.005215423, 0.014576883, -0.0035352057, -0.010624652, -0.039840713, -0.022439817, -0.014701472, 0.0053538554, 0.009344156, 0.010001708, 0.007835248, -0.011704421, 0.00010723068, -0.032310016, 0.0281709, -0.0006501975, 0.014673785, -0.0118082445, -0.01582277, 0.018106898, 0.019574277, 0.021996835, -0.0023966029, 0.0007246047, -0.009787139, 0.013330996, -0.020224907, -0.02131852, 0.010195512, -0.028655412, -0.033971198, -0.0045613325, -0.0033742785, 0.0043329196, 0.0009041336, 0.03156248, -0.011427557, -0.023824137, 0.0009828667, 0.021290833, -0.051302876, 0.002088592, 0.009558725, 0.01360786, -0.013725527, 0.029679809, 0.02119393, -0.015850456, 0.00728152, 0.0024450542, -0.035383202, 0.00015422184, 0.010001708, -0.023658019, 0.021166245, -0.036435287, -0.047343723, -0.015338259, -0.026426658, -0.0042533213, 0.030870322, 0.017442424, 0.015019866, 0.033472843, 0.017567012, -0.008901174, -0.02626054, -0.015144454, 0.034524925, 0.0205433, 0.021830717, 0.019338941, -0.011212987, 0.021664599, -0.015310572, 0.008368211, 0.04330151, -0.043938298, 0.01755317, 0.015172141, 0.019837297, -0.018300703, -0.013116427, -0.007350736, 0.0016741613, -0.008894253, 0.024114843, 0.01148293, 0.021636913, -0.01751164, -0.014936807, -0.0021578078, 0.015615123, -0.020820163, 0.028406234, 0.03748737, -0.034137316, -0.009212646, -0.027312621, 0.024682416, -0.036629092, -0.0128187975, -0.010417003, 0.01798231, 0.0012190663, 0.016362656, -0.024377864, 0.013753213, 0.039010122, -0.020363338, -0.037155133, -0.022813585, 0.025845243, -0.004409057, 0.029956672, -0.0024415934, -0.02214911, -0.011503695, 0.0018238408, 0.0057864552, -0.015656652, 0.0036822895, -0.034081943, 0.028184744, -0.039231613, -0.012714974, -0.014396922, 0.014175431, 0.009164195, -0.008617388, 0.024751632, 0.0046824603, -0.020294122, 0.0033119842, -0.041944876, 0.021941463, 0.026440501, 0.007814483, -0.014784532, -0.004242939, -0.01877137, 0.027561799, 0.02123546, -0.0011732107, -0.018273016, 0.030981068, 0.0012337747, -0.0009880579, 0.0014154665, -0.025872929, -0.0010330484, 0.024543984, -0.0044748127, 0.029236825, 0.033832766, 0.0030057034, -0.008894253, -0.0034002345, -0.009226489, -0.0018324928, -0.027298778, -0.03770886, 0.023907196, -0.016265754, -0.01412698, 0.04205562, 0.013469428, -0.007028882, 0.021290833, 0.01997573, 0.0068419985, 0.005378081, -0.02149848, -0.019103607, -0.024308648, 0.0019103608, 0.009157273, 0.012901857, 0.0032341161, 0.027381837, 0.039536163, 0.039314672, 0.0020245672, 0.012389659, 0.03610305, 0.013981626, 0.042913903, -0.03995146, 0.025111554, 0.00979406, 0.022730524, 0.034386493, 0.015767397, -0.020681731, -0.073258184, -0.021221617, 0.005772612, 0.042830843, -0.010320101, 7.402864e-05, 0.0059006615, -0.020307966, -0.0017009825, 0.05290869, 0.01863294, -0.029873613, -0.007163853, -0.035050966, 0.0065547526, -0.01742858, 0.023907196, -0.015421318, -0.034746416, 0.019740395, 0.010029394, 0.0145353535, 0.021097029, -0.018522194, 0.018259173, 0.04966938, -0.014549197, -0.008991155, -0.042083308, 0.019864984, -0.018979019, 0.017539326, -0.01092228, 0.034441866, 0.008928861, 0.04966938, -0.03388814, -0.029209139, 0.026731208, -0.027146503, 0.0043536844, 0.030925695, 0.053600848, 0.01387088, 0.004042213, -0.0016758917, 0.004308694, -0.016736422, -0.023436528, -0.041640326, 0.030261222, -0.01863294, 0.011413714, 0.0033016019, 0.037570428, -0.005973338, -0.011635205, 0.010520827, -0.046706937, 0.02851698, -0.018259173, -0.016279597, -0.003211621, 0.0006986487, -0.015102925, 0.04263704, 0.02180303, 0.0012216619, 0.002327387, 0.014853748, -0.02882153, 0.027077287, 0.032337703, 0.011060712, -0.0052881003, -0.0023325782, 0.024336334, -0.014784532, 0.013254859, 0.03280837, 0.018162271, 0.02271668, -0.021775344, 0.03081495, 0.015490534, 0.01712403, -0.004142576, 0.007350736, 0.07237222, 0.010153983, 0.027243406, 0.014922963, 0.04008989, 0.028766157, -0.003289489, -0.0064267027, 0.009607176, 0.022841271, -0.015476691, -0.0017909632, 0.035189398, 0.00080766383, 0.0092334105, -0.022744367, -0.06412168, -0.0021508862, 0.017151717, 0.0088458005, 0.0015115037, -0.014867591, 0.03948079, -0.024114843, -0.002154347, -0.0046097836, 0.014922963, -5.439943e-05, 0.029845927, 0.033472843, 0.0102093555, -0.029679809, -0.042083308, -0.023256566, -0.013683997, -0.0010840951, -0.042083308, -0.013573252, -0.0017996152, -0.003398504, -0.008811193, 0.008804271, 0.013725527, 0.015933516, 0.0039695357, 0.0038380255, -0.010382395, 0.02184456, -0.027451053, 0.00931647, 0.016805638, 0.017345522, 0.03831796, 0.0051773544, -0.0037999568, -0.005135825, 0.008361289, -0.016764108, 0.0021353127, 0.021373892, 0.037874978, 0.02552685, 0.032697625, 0.002137043, 0.016224224, 0.029790554, 0.02743721, 0.010929202, 0.0055753463, -0.0065928213, -0.013760135, -0.026108263, -0.012597307, -0.0072053825, -0.00554766, 0.028932275, 0.012638836, 0.047316037, -0.006786626, -0.0017459729, -0.028932275, -0.018563723, -0.030233536, -0.002439863, 0.022688994, -0.025319202, 0.009738687, -0.014992179, 0.037459683, 0.018715998, -0.026938856, 0.0370167, 0.029624436, -0.023505744, -0.0014275794, -0.014853748, -0.0026527022, -0.024640886, 0.013863958, -0.008478956, 0.005682631, -0.0047828234, -0.009434137, -0.011752872, 0.024031784, 0.0037999568, -0.013178721, 0.015601279, 0.0153936315, -0.033472843, 0.0034711808, -0.008430505, -0.035826188, -0.0050216187, 0.02652356, 0.013891645, 0.040754363, 0.031839345, -0.014154666, -0.030095104, -0.004685921, 0.0020038025, -0.02601136, -0.034081943, -0.0047170683, 0.0037065153, -0.015435161, 0.0059975637, 0.019034391, -0.008928861, -0.023561116, 0.027409524, 0.029098393, -0.023394998, -0.03709976, 0.013767056, -0.026163638, 0.0015045821, -0.013600938, -0.005499209, -0.039813027, 0.051828917, -0.00034759394, -0.007876777, 0.054819047, -0.016930226, 0.01308874, -0.03726588, 0.022038365, -0.0065547526, 0.028876903, 0.035770815, -0.02080632, -0.048035882, 0.0019242039, 0.0053365515, -0.041723385, 0.04460277, -0.010631573, -0.006437085, -0.010153983, 0.016847167, 0.023810294, -0.021706128, -0.01126836, -0.007198461, 0.13178721, 0.0017606813, 0.00084486743, -0.0033604354, 0.018494507, -0.009828668, -0.005513052, -0.022356758, -0.0044055963, 0.017110188, -0.025609909, 0.007814483, -0.008894253, 0.013670154, -0.014355392, 0.0063505652, -0.0022893182, -0.05825216, 0.0060321717, -0.046956114, -0.017843878, 0.022398287, 0.021609226, 0.018369919, 0.0058314456, -0.03410963, -0.011247595, -0.03610305, 0.031977776, 0.021124715, 0.00015941303, -0.006011407, -0.029458316, 0.008125955, -0.012071265, -0.033140607, -0.009877119, 0.039425418, -0.007246912, -0.018397605, 0.0054507577, 0.012998759, 0.008707369, -0.009655627, 0.029569063, 0.008991155, 0.0027149965, 0.0018082672, 0.01456304, 0.02691117, -0.02241213, -0.00035300144, 0.0005186872, -0.008506643, 0.014050842, -0.05290869, 0.012230461, -0.022301385, 0.034774102, -0.02106934, -0.0046616956, -0.018300703, 0.030759577, 0.0075860703, -0.014230804, -0.024627043, -0.0077867964, -0.005232727, 0.012964151, -0.016694892, -0.00020613382, 0.001044296, -0.008790428, -0.03280837, -0.015379788, -0.030981068, 0.04355069, 0.014895277, -0.029541377, -0.009434137, 0.020377181, 0.019089764, 0.031590167, -0.0046963035, -0.0070011956, 0.0034140777, -0.007509933, -0.032919116, -0.017151717, -0.021110872, 0.013773978, 0.002695962, 0.0053503946, 0.033085234, 0.003713437, 0.00057146436, -0.009759452, 0.0060702404, -0.014756844, -0.041889504, 0.030565772, -0.005703396, -0.014092372, 0.043495316, 0.012528091, -0.0205433, -0.030981068, -0.022135267, 0.016542617, 0.010520827, 0.018079212, -0.0043813707, 0.014043921, -0.021941463, 0.013455585, -0.005118521, -0.04714992, 0.018065369, -0.01326178, -0.021041654, -0.008942704, 0.01798231, -0.0006493323, 0.02080632, -0.009711001, -0.0057207, -0.01278419, 0.017650072, -0.0036130736, 0.00065928214, 0.00953796, -0.048977222, 0.0059802597, -0.013704762, 0.018231487, -0.0039210846, 0.019255882, -0.04460277, 0.030288909, 0.02119393, 0.026634306, 0.008029053, -0.021913776, -0.007745267, 0.0053850026, -0.009641784, 0.005270796, 0.0060356325, -0.0036926721, -0.028226273, -0.041197345, 0.0027184573, -0.012798033, -0.0047136075, 0.016611833, -0.0039003198, -0.0103270225, -0.0036303776, -0.010043237, 0.005945652, -0.014452294, -0.0018463361, 0.0069146752, 0.04468583, 0.022702837, -0.010624652, -0.0070323427, 0.026343599, 0.016265754, -0.0007233069, -0.017345522, -0.043523002, 0.0025644517, -0.006305575, 0.02375492, -0.014002391, -0.01027165, 0.0071500097, 0.001014014, -0.0026371286, 0.012631915, -0.005066609, -0.020681731, 0.0076968158, -7.759759e-05, 0.011351419, 0.0039937613, -0.011683656, -0.020750947, -0.0382349, 0.031174872, -0.012922621, 0.052216526, 0.0075653056, -0.010590044, 0.018605253, -0.031783972, -0.039619222, -0.010001708, -0.010846143, 0.005270796, -0.01832839, -0.006949283, 0.029209139, 0.006869685, 0.0070323427, -0.030288909, 0.012202775, -0.014133901, 0.006578978, 0.013420977, -0.0050596874, -0.012694209, 0.009745609, 0.022301385, 0.00580722, 0.019615807, 0.005052766, -0.031479422, -0.0019864982, 0.009544882, 0.004107968, 0.02552685, -0.020487927, -0.0009127856, -0.01304721, 0.03363896, -0.040062204, 0.0020436016, -0.04543336, 0.004156419, 0.033666648, 0.0010927471, -0.035632383, -0.023242723, 0.0030316594, -0.015185984, 0.0039072414, -0.0015149645, 0.008568937, -0.0155459065, 0.0056757093, -0.03280837, 0.028987648, 0.028073998, -0.0054749832, 0.01187746, 0.009108822, 0.006904293, -0.009115743, 0.011912068, -0.0067450963, -0.023741078, 0.0075030113, 0.0028430459, -0.010160904, 0.03671215, -0.032310016, -0.0065409094, -0.020252593, -0.026966542, -0.005198119, 0.003415808, 0.0015504378, 0.047620587, 0.011704421, 0.015144454, 0.007572227, -0.028406234, -0.0020834007, -0.031008754, -0.005942191, 0.003606152, 0.02465473, 0.015089082, 0.020003416, 0.01789925, 0.00420487, 0.015684338, -0.031479422, 0.007350736, -0.03350053, -0.00053426076, -0.011545224, -0.009441058, 0.015933516, 0.0048001274, 0.020391025, -0.007426874, 0.006596282, -0.015075238, -0.0055926503, 0.0066551156, 0.019338941, 0.012175089, 0.006250202, 0.005326169, -0.030399654, -0.052327275, 0.0040629776, -0.0019709247, -0.0305104, 0.016155008, -0.009738687, -0.0077175805, -0.027617171, 0.03784729, -0.026592776, 0.0058660535, 0.0086104665, -0.009371842, -0.012382737, -0.0062813493, -0.0020003414, -0.016390342, -0.005018158, -0.0042152526, 0.0071361666, -0.017622385, -0.008679682, -0.033362098, 0.025872929, -0.018743685, -0.021512324, -0.004672078, 0.032531507, -0.015310572, -0.002978017, 0.0020695576, -0.0027617172, -0.01629344, 0.017027128, -0.023561116, -0.0035750049, 0.056757096, 0.00054594094, -0.016182695, -0.010389317, -0.0076206783, -0.030953381, -0.02184456, -0.0029745563, 0.0012259879, 0.008347446, 0.0018844048, 0.029486002, 0.007246912, 0.0017978848, 0.022550562, 0.021304676, -0.033168294, -0.020667888, 0.041141972, -0.03424806, -0.030538086, -0.014826061, 0.035853874, 0.0014621874, -0.03956385, 0.00029524937, 0.026897326, 0.011372183, 0.023547273, -0.0036130736, 0.0030178162, 0.0078006396, -0.018522194, 0.008388976, -0.005250031, -0.0011161076, -0.05764306, 0.035134025, 0.0006934575, -0.03219927, 0.007862934, 0.004862422, -0.010866907, 0.0033344794, -0.05149668, -0.013718605, -0.00042351522, -0.023464214, 0.03665678, -0.002894958, -0.023907196, 0.006056397, -0.001559955, -0.02241213, -0.019920357]

# From this PR
llm.embed(text: "test").embedding.count
# => 512
print llm.embed(text: "test").embedding
# => [-0.014155505, 0.0021972281, 0.02241867, -0.07848518, -0.009180743, -0.01855988, 0.013847993, -0.0194229, 0.04110751, 0.011268853, 0.045591243, -0.009379138, 0.0055302694, 0.014552296, 0.020147044, 0.06400233, -0.08570677, -0.0034967172, -0.073247544, 0.05233668, 0.051741496, 0.03499693, 0.057256885, -0.062415164, 0.049519468, -0.028033257, -0.020037927, 0.0152169205, 0.044718303, -0.059558272, 0.08427833, -0.04138526, -0.0019554337, -0.055511005, 0.07908037, 0.0059320196, 0.03378672, 0.027021442, -0.008124288, -0.0056840256, -0.059399556, -0.06451815, 0.016268415, 0.039064035, 0.0430121, -0.026922243, -0.049281396, -0.05543165, 0.06951772, 0.039936975, -0.04555156, -0.0039827856, 0.06912093, 0.1006261, 0.027695986, 0.03428271, -0.050432086, 0.028549084, 0.026584972, 0.0076481393, -0.003290882, -0.0270016, 0.055034857, 0.0078911735, -0.045829315, -0.086103566, 0.017538145, 0.045115095, 0.0006302151, 0.03085047, 0.06582756, 0.017032238, -0.018361486, 0.024005832, 0.0125088245, 0.028648281, 0.0038687084, 0.026108822, 0.020891026, -0.011536687, 0.028965715, 0.015742667, -0.028172133, 0.013738875, 0.044678625, -0.018440844, -0.12975053, 0.008957548, 0.009885047, -0.052297004, -0.016020421, -0.005108679, 0.02194252, 0.07408081, 0.03344945, -0.039401308, -0.0036554337, -0.02577155, 0.0034917574, -0.010395914, 0.014949086, 0.01393727, -0.0019479939, 0.025692193, 0.032199558, 0.026743688, -0.045432527, 0.03380656, -0.017885337, 0.047654554, -0.008407001, -0.029541062, -0.025156526, 0.04221852, 0.064121366, 0.021644928, 0.15736715, -0.1483997, 0.04864653, -0.10221326, 0.039202914, 0.043726325, 0.017974615, -0.032735225, 0.04531349, -0.030136248, -0.008684754, -0.013540479, -0.032854263, 0.00012508204, -0.045472205, 0.007072793, 0.0473768, 0.013719035, -0.0154946735, -0.011953318, -0.036048427, 0.006452807, -0.084040254, 0.05959795, 0.016208896, 0.034401745, 0.009141064, 0.00072600285, -0.02327177, -0.040452804, -0.02408519, 0.057733033, 0.03265587, 0.040413123, -0.01644697, 0.008094529, -0.011467249, 0.0052673956, -0.0769377, -0.05412224, -0.022617064, -0.028132455, -0.04487702, -0.0024613417, 0.0346795, -0.007152151, 0.016198976, -0.03644522, -0.036623772, 0.012320349, -0.011298613, -0.025077166, -0.07554893, 0.027636467, 0.038250614, 0.022240113, 0.019541938, -0.01778614, -0.029541062, 0.034401745, -0.0321202, 0.014879648, -0.032060683, 0.00296353, -0.028807, 0.072811075, -0.036623772, 0.027021442, 0.021744126, -0.084754474, 0.03465966, 0.007653099, 0.004320058, -0.019819692, -0.008679795, -0.013461121, 0.04186141, 0.054717425, 0.039639384, -0.1108633, 0.026465934, 0.044242155, -0.01949234, 0.05840758, -0.050392408, -0.0042406996, 0.014711012, -0.016466811, 0.021724286, 0.019512178, -0.034302548, 0.067454405, -0.024938291, -0.035790514, 0.09999123, -0.0020137124, -0.0027824943, 0.0041985405, -0.007856454, 0.010842304, 0.072969794, 0.01812341, -0.012598102, 0.087532006, -0.011576367, 0.014472938, 0.12586199, 0.017548066, -0.014631654, 0.11975141, 0.01605018, -0.02033552, 0.001988913, -0.08090561, -0.061542224, 0.025712032, 0.010961341, -0.0060609765, -0.03418351, 0.037437193, -0.048051342, -0.0059716986, -0.050074976, -0.073882416, -0.053884167, 0.013619838, -0.042496275, -0.056145873, -0.029719617, -0.018668998, 0.03555244, -0.013778554, 0.03678249, 0.021783805, 0.007474543, -0.05376513, -0.05368577, -0.098721504, 0.039143395, -0.015078044, 0.036504738, 0.0035760754, 0.029997371, -0.011645804, -0.07336658, -0.0051830774, -0.060867682, -0.071382634, -0.0947536, 0.0074646235, 0.042020127, -0.020434717, 0.019214585, -0.063684896, -0.009527935, 0.021684607, 0.02662465, -0.03003705, -0.04475798, 0.059121802, -0.0017458787, 0.027120639, 0.03717928, 0.038369652, -0.060828, -0.01943282, -0.007013274, -0.040829755, -0.016169218, 0.034600142, 0.015712908, -0.065708525, 0.010286797, 0.08729394, -0.018351566, -0.0155740315, 0.001859956, 0.047178403, 0.023192411, 0.03283442, 0.029064912, -0.0574156, 0.00075948203, -0.08197694, 0.045353167, 0.0432105, 0.0076233395, 0.040710717, -0.013421442, 0.06495462, 0.019750252, -0.011457329, 0.005242596, -0.023866955, 0.011992997, 0.028172133, -0.0011686723, 0.010753025, 0.02073231, -0.073009476, -0.018976511, 0.082135655, 0.05166214, -0.03255667, 0.017934935, -0.032417793, 0.048844926, 0.046464182, -0.022299632, -0.003851349, 0.0021935082, -0.03842917, 0.020494236, 0.047535516, -0.13673404, -0.040075853, -0.00506652, -0.0122509105, 0.05067016, 0.046226107, 0.013252807, 0.07943748, -0.008392122, 0.059796344, -0.09602333, 0.016228735, -0.0041043027, -0.01731991, 0.0659466, 0.032437634, 0.064795904, -0.034461264, 0.044162795, 0.02033552, 0.06174062, 0.015207, -0.0031966444, 0.030830631, 0.030810792, -0.07293011, -0.024005832, 0.021387015, 0.046662576, -0.0025716992, 0.020226402, 0.061621584, 0.06463719, 0.014760611, 0.06713697, 0.028568923, -0.038091898, -0.016278336, -0.020831507, 0.0013366884, 0.014869728, 0.049717866, 0.065272056, 0.024640698, -0.049717866, 0.043766003, 0.06130415, 0.09951509, 0.033112176, 0.06558949, 0.04067104, 0.013887672, 0.050471768, -0.015831945, 0.040452804, 0.0035983948, -0.02243851, 0.038746603, -0.005041721, -0.028430048, 0.073723696, -0.02914427, -0.024501821, -0.10951421, 0.09062698, 0.03598891, 0.027120639, 0.11411698, 0.026108822, -0.0038315093, 0.03210036, -0.056383945, -0.059796344, 0.05793143, -0.004312618, 0.013153609, -0.046146747, -0.019879209, 0.046662576, -0.0037149522, -0.03763559, 0.047932304, 0.047099046, -0.0029660098, -0.018063894, -0.03247731, 0.043845363, -0.005396352, 0.0456706, 0.01565339, -0.04741648, -0.054757103, -0.016814003, 0.02204172, -0.032060683, 0.0023125452, 0.06459751, 0.02704128, 0.050392408, -0.032993138, -0.0033057618, -0.028549084, -0.051424064, 0.053050905, -0.04904332, -0.02331145, 0.010485192, 0.00549555, -0.0074795033, 0.019581616, -0.034520783, -0.0073406263, 0.021863163, -0.031167902, -0.05971699, 0.02660481, -0.050074976, -0.060629606, 0.012350108, -0.028092775, 0.015107803, -0.050947916, -0.01940306, -0.085786134, 0.022557547, -0.07717577, 0.01733975, -0.0048036464, 0.09308708, 0.025930267, 0.052495398, 0.06602596, 0.02243851, 0.06923996, 0.07316819, -0.008635156, 0.037377674, 0.049638506, 0.061185114, -0.057455283, -0.012598102, -0.028608603, 0.06618468, 0.0071670306, -0.050392408, -0.021248138, 0.034937415, -0.005118599, 0.0010217358, -0.11260918, -0.045948353, -0.047575194, -0.04432151, -0.023132892, 0.017171115, 0.1641126, -0.05491582, 0.0017409188, -0.027973738, -0.021684607, 0.047575194, 0.0012195111, 0.033429608, -0.022299632, 0.008689715, 0.04868621, 0.039460827, 0.028826838, -0.0014433259, 0.011249014, -0.10380042, -0.03422319, -0.05070984, 0.09618205, 0.036584094, 0.02920379, -0.01564347, -0.026704008, 0.07233493, -0.004453975, -0.025255723, 0.004984682]

```

Reply:
> One issue might be when models support multiple dimension sizes, we would need to use the passed dimension, rather than default_dimension when calling create_default_schema to create the vector store collection.

I think in that case we'd just have something like:
```ruby
    def default_dimension
        @defaults[:dimension] || EMBEDDING_SIZES.fetch(defaults[:embeddings_model_name].to_sym)
    end
```

So it's whatever you passed in OR the dimension size for the default embeddings model.